Libraries/LibWeb/HTML/Dates.cpp

```diff
@@ -462,24 +443,40 @@ static Optional<HourMinuteSecond> parse_a_time_component(GenericLexer& input)
         if (second_string.find_all("."sv).size() > 1)
             return {};
         // Otherwise, interpret the resulting sequence as a base-ten number (possibly with a fractional part). Set second
-        // to that number.
-        // NB: The spec doesn't state requirements for what we must do with the fractional part of the second(s) string.
-        //     The spec neither requires that we separately preserve it nor requires that we completely discard it. If we
-        //     did have any reason at all to preserve it, we could parse the string into a float here. But there doesn't
-        //     seem to be any point in doing that, because there’s nothing in the corresponding calling algorithm(s) in the
-        //     spec that takes a milliseconds field and does anything with it anyway.
-        auto maybe_second = second_string.to_number<i32>();
+        // to that number..
+        auto maybe_second = second_string.to_number<f32>();
         if (!maybe_second.has_value())
             return {};
         second = maybe_second.value();
 
         // 4. If second is not a number in the range 0 ≤ second < 60, then fail.
-        if (second < 0 || second > 60)
+        if (second < 0 || second >= 60)
             return {};
     }
 
     // 8. Return hour, minute, and second.
-    return HourMinuteSecond { hour, minute, second };
+    return HourMinuteSecond { hour, minute, (f32)second };
```

Comment:
Do we need this cast? Isn't it already declared an `f32`?
If I'm missing something and this is needed, we use `static_cast` instead of c-style casts.